src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.Update.cs

```diff
@@ -289,17 +289,45 @@ private async Task<TimeSpan> DownloadFullDatabaseAsync(FileInfo databaseFileInfo
             // Write the file out to disk so we'll have it the next time we launch VS.  Do this
             // after we set the in-memory instance so we at least have something to search while
             // we're waiting to write.
-            await WriteDatabaseFileAsync(databaseFileInfo, bytes, cancellationToken).ConfigureAwait(false);
+            await WriteDatabaseTextFileAsync(databaseFileInfo, bytes, cancellationToken).ConfigureAwait(false);
+            await WriteDatabaseBinaryFileAsync(database, databaseFileInfo, cancellationToken).ConfigureAwait(false);
```

Comment:
there's a part of me that is concerned about torn writes.  specifically that one of these succeeds and the other does not.  leading to an inconsistent state.

Reply:
I think it's OK for the txt file to be written out and the binary file to not be. The subsequent VS session will just read from the txt file and again try to persist out the binary file.

Reply:
what about the reverse?  or do these functions fail instead of continuing on?  if this is the case, doc that it's ok to fail to write this file.

Reply:
Looking at this a little closer, this is tricky and I don't know if it works the way it's intended currently. The RepeatIOAsync doesn't appear to throw if it fails the maximum number of times. I'm hesitant to change it as maybe I'm not understanding the exception flow through that call.

Reply:
After a bit more consideration, I think it's ok if the bin file is persisted and the txt file isn't. The next VS run will not find the txt file and will go through the DownloadFullDatabaseAsync path in UpdateDatabaseInBackgroundWorkerAsync and use the newly downloaded file to populate the db rather than the bin file.